test/ContractDeployment.ts

```diff
@@ -0,0 +1,11 @@
+import {ethers} from "hardhat";
+import {expect} from "chai";
+
+describe("Staking contract deployment", async () => {
+    it("Min validators should not be greater than max validators number", async function () {
+        const contractFactory = await ethers.getContractFactory("Staking");
+        await expect(contractFactory.deploy(7, 6)).to.be.revertedWith(
+            "Min validators num can not be greater than max num of validators"
+        );
+    })
+})
```

Comment:
Why new file?

Reply:
No particular reason, I think it is clearer.
In test/ContractDeployment.ts we can write all tests for deployment (if we add any more conditions)
in test/Staking.ts we can test all edge cases for staking/unstaking.

If you do not agree I do not agree we can merge them, but I would not like to have 1 test file for the whole repo.

Reply:
I think we should separate by Contract file (implementation). It is possible to merge test file?

Reply:
Yes of course, fixed 90257b0ef518f9c4e5ebd09bd6999a842b1dc26f